recidiviz/ingest/generic_scraper.py

```
@@ -0,0 +1,620 @@
+        if task_type & constants.GET_MORE_TASKS:
+            tasks = self.get_more_tasks(content, params)
+            for task_params in tasks:
+                # Always pass along the scrape type as well.
+                task_params['scrape_type'] = params['scrape_type']
+                self.add_task('_generic_scrape', task_params)
+        if task_type & constants.SCRAPE_PERSON:
+            params['person'] = self._populate_person(content, params)
+        if task_type & constants.SCRAPE_RECORD:
+            record = self._populate_record(content, params)
+            self._populate_snapshot(record)
+
+    def _populate_person(self, content, params):
+        """
+        Calls all of the necessary functions to populate a person in the database
+
+        Returns:
+            A person database object that is passed through to future functions.
+        """
+        person_id = self.get_person_id(content, params)
+        person = self.get_person_class().get_or_insert(person_id)
+        person.person_id_is_fuzzy = self.person_id_is_fuzzy()
+        person.given_names = self.get_given_names(content, params)
+        person.surname = self.get_surname(content, params)
+        person.suffix = self.get_suffix(content, params)
+        person.alias = self.get_alias(content, params)
+        person.birthdate = self.get_birthdate(content, params)
+        person.age = self.get_age(content, params)
+        person.region = self.get_region().region_code
+        person.sex = self.get_sex(content, params)
+        person.race = self.get_race(content, params)
+        # Lastly populate any extra params that may be scraper specific.
+        self.populate_extra_person_params(content, params, person)
+
+        try:
+            person.put()
+        except (Timeout, TransactionFailedError, InternalError):
+            # Datastore error - fail task to trigger queue retry + backoff
+            logging.warning("Couldn't persist person: %s", person.person_id)
```

We should expose some information from the exception on this one instead of swallowing it all. Same goes for the other catches in this class.

---

Done.